Jobs/Statistics/LoadMetricsDataJob.php

```diff
@@ -0,0 +1,112 @@
+            if (!file_exists($archivedFilePath)) {
+                $filename .= '.gz';
+                $archivedFilePath = StatisticsHelper::getUsageStatsDirPath() . DIRECTORY_SEPARATOR . FileLoader::FILE_LOADER_PATH_ARCHIVE . DIRECTORY_SEPARATOR . $filename;
+            }
+            $stagingPath = StatisticsHelper::getUsageStatsDirPath() . DIRECTORY_SEPARATOR . FileLoader::FILE_LOADER_PATH_STAGING . DIRECTORY_SEPARATOR . $filename;
+
+            if (!rename($archivedFilePath, $stagingPath)) {
+                $message = __('admin.job.loadMetricsData.returnToStaging.error', ['file' => $filename,
+                    'archivedFilePath' => $archivedFilePath, 'stagingPath' => $stagingPath]);
+            } else {
+                $message = __('admin.job.loadMetricsData.error', ['file' => $filename]);
+            }
+            $this->failed(new JobException($message));
+            return;
+        }
+
+        $statsTotalDao = DAORegistry::getDAO('UsageStatsTotalTemporaryRecordDAO'); /* @var UsageStatsTotalTemporaryRecordDAO $statsTotalDao */
+        $statsUniqueItemInvestigationsDao = DAORegistry::getDAO('UsageStatsUniqueItemInvestigationsTemporaryRecordDAO'); /* @var UsageStatsUniqueItemInvestigationsTemporaryRecordDAO $statsUniqueItemInvestigationsDao */
+        $statsUniqueItemRequestsDao = DAORegistry::getDAO('UsageStatsUniqueItemRequestsTemporaryRecordDAO'); /* @var UsageStatsUniqueItemRequestsTemporaryRecordDAO $statsUniqueItemRequestsDao */
+        $statsInstitutionDao = DAORegistry::getDAO('UsageStatsInstitutionTemporaryRecordDAO'); /* @var UsageStatsInstitutionTemporaryRecordDAO $statsInstitutionDao */
+
+        $statsTotalDao->deleteByLoadId($this->loadId);
+        $statsUniqueItemInvestigationsDao->deleteByLoadId($this->loadId);
+        $statsUniqueItemRequestsDao->deleteByLoadId($this->loadId);
+        $statsInstitutionDao->deleteByLoadId($this->loadId);
+    }
+
+    /**
+     * Load the entries inside the temporary database associated with
+     * the passed load id to the metrics tables.
+     */
+    private function _loadData(): bool
+    {
+        $statsTotalDao = DAORegistry::getDAO('UsageStatsTotalTemporaryRecordDAO'); /* @var UsageStatsTotalTemporaryRecordDAO $statsTotalDao */
+        $statsUniqueItemInvestigationsDao = DAORegistry::getDAO('UsageStatsUniqueItemInvestigationsTemporaryRecordDAO'); /* @var UsageStatsUniqueItemInvestigationsTemporaryRecordDAO $statsUniqueItemInvestigationsDao */
+        $statsUniqueItemRequestsDao = DAORegistry::getDAO('UsageStatsUniqueItemRequestsTemporaryRecordDAO'); /* @var UsageStatsUniqueItemRequestsTemporaryRecordDAO $statsUniqueItemRequestsDao */
+
+        $statsTotalDao->removeDoubleClicks(StatisticsHelper::COUNTER_DOUBLE_CLICK_TIME_FILTER_SECONDS);
+        $statsUniqueItemInvestigationsDao->removeUniqueClicks();
+        $statsUniqueItemRequestsDao->removeUniqueClicks();
```

Comment:
I will leave it as it is, because those are clicks counting to one unique click i.e. all except one will be removed.

Reply:
When I see the method name, `removeUniqueClicks()` it sounds like it will remove entries that are unique. For example, if someone only visited a submission once, it would remove that one. 

Maybe call it something like `compileUniqueClicks()` or something? Other options could be `pruneDuplicateClicks()` or `removeDuplicateClicks()`.

Reply:
OK, I will call it compileUniqueClicks... :+1: